new datastruct workshop (#217)
NuevoFoundation/workshops: content/english/datastruct/Arraylist/problem 1.md

---
title: "Problem 1: Removing and looping"
date: 2020-02-10T13:24:17-07:00
draft: false
weight: 3
---

<p style="text-align: center;"><iframe width="60%" height="600px" src="https://www.youtube.com/embed/unwrTbTILmA" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe></p>

<link rel="stylesheet" href="../style.css">

> ##ArrayList Task 1: Inserting

Reviewer: ```suggestion
## ArrayList Task 1: Inserting
```

<p>Tacos Truck is now avaiable down the street! New employees are struggling to get keep track of all the different orders. Help them organize all the items using ArrayLists! </p>

```java
    /*
    Dan is ordering from his favorite taco shop:
        -2 orders of "carne asada"
        -4 orders of "carnitas"
        -1 order of "pollo"
        -2 orders of "birria"

        Return an arraylist of all of these elements in that order
    */

```

{{%notice tip%}}
1. How can you add items onto the list?
2. What does Dan want in the order?
{{%/notice%}}

<iframe frameborder="0" width="100%" height="800px" src="https://replit.com/@nuevofoundation/ListBasic?lite=true"></iframe>

## ArrayList Task 2: Getting Element

> <p>A restaurant selling Chinese food has just been opened nearby. A huge serious of orders has just come in and the manager has trouble keeping track of everything so he put it all in an ArrayList. Help finish the program to return the given string at a given index.</p>

Reviewer: ```suggestion
<p>A restaurant selling Chinese food has just opened nearby. A huge number of orders has just come in and the manager has trouble keeping track of everything, so they have put it all in an `ArrayList`. Help finish the program to return the given string at a given index.</p>
```

```java

    ArrayList<String> menu = new ArrayList<>(); 
  
        menu.add("Pizza"); 
        menu.add("Hotdog"); 
        menu.add("Hamburger"); 
        menu.add("Hotdog"); 

    //if we run the code
    item = find(menu, 0);

    //find will return since it is the 0th item in the menu
    item = "Pizza"

```

{{%notice tip%}}
1. How can you iterate through the list?
2. Look at the example menu!
{{%/notice%}}

<iframe frameborder="0" width="100%" height="800px" src="https://replit.com/@nuevofoundation/ListGet?lite=true"></iframe>

## ArrayList Task 3: Removing Elements

> <p>The same Chinese restaurant has a bug in their code! Orders are randomly duplicating and the ArrayList are filled with copies. Help by writing a program to remove the first n occurrences of the given order.</p>

Reviewer: ```suggestion
<p>The same Chinese restaurant has a bug in their code! Orders have been duplicated randomly and the `ArrayList` are filled with copies of orders. Help the owner out by writing a program to remove the first `n` occurrences of a given order in the `ArrayList`.</p>
```
